pmlc/dialect/pxa/transforms/passes.h

```diff
@@ -13,10 +15,24 @@ class OpPassBase;
 
 namespace pmlc::dialect::pxa {
 
+struct StencilPassOptions
+    : public mlir::PassPipelineOptions<StencilPassOptions> {
+  Option<int> numberOfThreadsOption{
+      *this, "affine-stencil-number-threads",
+      llvm::cl::desc("Specifies number of threads for the stencilling pass")};
+};
+
 struct StencilPass : public mlir::FunctionPass<StencilPass> {
+  StencilPass() = default;
```

Comment:
I don't think we should allow an 'unconfigured' pass. Why not force the number of threads to be specified always?

Reply:
I can take it. If you don't instantiate the pass with explicit option value, the default is created, which is zero - or option not specified. SOeone might just want to let the RT of the OS or the stenciling pass decide what to do with that. I think that is how people will use it in most of the cases, but you still have an option to override the default.

Reply:
My feeling is that we should prefer explicit over implicit. Configuration of passes should all be explicitly defined someplace so that we don't 'accidentally' run into good or bad behavior.

Reply:
Taken out.